modules/authentication/src/handlers/local.ts

```diff
@@ -545,6 +572,89 @@ export class LocalHandlers implements IAuthenticationStrategy {
     return 'Password changed successfully';
   }
 
+  async changeEmail(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
+    const { newEmail, password } = call.request.params;
+    const { user } = call.request.context;
+    const config = ConfigController.getInstance().config;
+
+    const validation = await this.validate();
+    if (!validation) {
+      throw new GrpcError(status.INTERNAL, 'Cannot use verification at this time');
+    }
+    if (user.email === newEmail) {
+      throw new GrpcError(
+        status.INVALID_ARGUMENT,
+        'The new email can not be the same as the old email',
+      );
+    }
+    if (AuthUtils.invalidEmailAddress(newEmail)) {
+      throw new GrpcError(status.INVALID_ARGUMENT, 'Invalid email address provided');
+    }
+    const dbUser: User | null = await User.getInstance().findOne(
+      { _id: user._id },
+      '+hashedPassword',
+    );
+    if (isNil(dbUser)) {
+      throw new GrpcError(status.UNAUTHENTICATED, 'User does not exist');
+    }
+    if (isNil(dbUser.hashedPassword)) {
+      throw new GrpcError(
+        status.PERMISSION_DENIED,
+        'User does not use password authentication',
+      );
+    }
+    const passwordsMatch = await AuthUtils.checkPassword(
+      password,
+      dbUser.hashedPassword!,
+    );
+    if (!passwordsMatch) {
+      throw new GrpcError(status.UNAUTHENTICATED, 'Invalid password');
+    }
+    const dupEmailUser = await User.getInstance().findOne({ email: newEmail });
```

Comment:
make this check after the check of line 590, before checking password validation. In  general you should try and keep a parameters validation in the same place and not in various locations throughout the code.